src/main/java/de/tum/in/www1/artemis/service/scheduled/AutomaticSubmissionService.java

```diff
@@ -119,10 +119,13 @@ private Submission updateParticipation(Submission submission) {
             if (submission instanceof ModelingSubmission) {
                 ModelingExercise modelingExercise = (ModelingExercise) exercise;
                 ModelingSubmission modelingSubmission = (ModelingSubmission) submission;
-                // notify compass about new submission
-                modelingSubmissionService.notifyCompass(modelingSubmission, modelingExercise);
-                // check if compass could assess automatically
-                modelingSubmissionService.checkAutomaticResult(modelingSubmission, modelingExercise);
+                try {
+                    // notify compass about new submission
+                    modelingSubmissionService.notifyCompass(modelingSubmission, modelingExercise);
+                }
+                catch (Exception ex) {
+                    log.error("Exception while notifying Compass about a new (automatic) submission:\n{}", ex.getMessage(), ex);
```

Comment:
Is it really ok to just log this exception? It seems to me like a you do this, because this method is only called by a cron job that should not fail when this issue happens.
But would it in this case still be ok that the participation is set to `FINISHED`?

Reply:
Yes, it should be okay to just log the exception here. Compass is not necessary for automatically submitting a submission. If Compass throws an Exception for whatever reason, the submission should get submitted anyway (save to database).
When tutors then query for submissions without assessment sooner or later also submissions will be returned that Compass does not know about.